### PR TITLE
Postfix 3.6.2

### PR DESCRIPTION
**Pull request by nlewo**
###### Motivation for this change

I new minor release has been published: http://www.postfix.org/announcements/postfix-3.6.2.html

- I updated the source code url because the current one didn't work. 
- As a nixos-mailserver maintainer, I added myself as postfix maintainer to better follow nixpkgs updates.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nlewo** [comment]
@GrahamcOfBorg test postfix

File: pkgs/servers/mail/postfix/default.nix [hunk 1]
```diff
@@ -103,6 +103,6 @@ in stdenv.mkDerivation rec {
     description = "A fast, easy to administer, and secure mail server";
     license = with licenses; [ ipl10 epl20 ];
     platforms = platforms.linux;
-    maintainers = with maintainers; [ globin dotlambda ];
+    maintainers = with maintainers; [ globin dotlambda nlewo ];
```
Comment (SuperSandro2000) [inline pkgs/servers/mail/postfix/default.nix:106]: You do not exist as a maintainer yet.

Comment (nlewo) [reply pkgs/servers/mail/postfix/default.nix]: Ah, it's actually my true pseudo and not my github one! 
Fixed. Thanks.

**Comment by SuperSandro2000** [comment]
@nlewo please resolve the merge conflict.

**Comment by nlewo** [comment]
@SuperSandro2000 conflicts are resolved.

**Comment by dotlambda** [comment]
> * I updated the source code url because the current one didn't work.

We should add `mirror://postfix`.

**Pull request merged**